multimodal/src/autogluon/multimodal/utils/load.py

```diff
@@ -138,3 +141,16 @@ def get_load_ckpt_paths(ckpt_path: str, dir_path: str, resume: bool):
             ckpt_path = None  # must set None since we do not resume training
 
     return load_path, ckpt_path
+
+
+def load_config_with_retry(file_path, retries=3, delay=2):
+    for attempt in range(retries):
+        try:
+            if os.path.exists(file_path):
+                return OmegaConf.load(file_path)
+            else:
+                print(f"Attempt {attempt+1}/{retries}: File {file_path} does not exist yet.")
+        except FileNotFoundError as e:
+            print(f"Attempt {attempt+1}/{retries}: FileNotFoundError: {e}")
+        time.sleep(delay)
+    raise FileNotFoundError(f"File {file_path} not found after {retries} retries.")
```

Comment:
1. Please add a detailed docstring for what this does as well as type hints.

2. Why are we using print statements? Shouldn't we be using `logger`?

3. Why is this in a try/except block? If we find the file via `os.path.exists`, why are we allowing for exceptions?

4. Why would we find the file in subsequent tries if we didn't find it originally? The logic doesn't quite make sense to me.

5. Even if we succeed we are calling time.sleep. This means even if we succeed the first try we are sleeping for 2 seconds without reason.

6. If we retried all times and failed each time, we still call time.sleep before raising the exception, which is unnecessary. time.sleep should only be called if the file wasn't found and there are retries remaining.

7. How is this change related to fixing the test_dump_hf unit test?